x-pack/test/functional/apps/discover/reporting.ts

```diff
@@ -142,8 +142,8 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
         expect(res.status).to.equal(200);
         expect(res.get('content-type')).to.equal('text/csv; charset=utf-8');
 
-        const csvFile = res.text;
-        expectSnapshot(csvFile).toMatch();
+        // const csvFile = res.text;
+        // expectSnapshot(csvFile).toMatch();
```

Comment:
@kertal Hey Matthias! 
I'm not really sure.  Would you like me to run something on my local?

Reply:
@wayneseymour I think this would be useful for the detective work, yes. many thx 🕵️ . you're ingesting the same data, right? Btw great performance improvement when this is done 🥳

Reply:
# Without Updating the Snapshot, but just uncommenting the snapshot check
Ok, so returning that test to it's original form and running it on my local results in a wall of text, but at the top of that is this:
```
         └- ✖ fail: discover Discover CSV Export Generate CSV: new search generates a report from a new search with data: default
         │      Error: expect(received).toMatchSnapshot()
         │
         │ Snapshot name: `discover Discover CSV Export Generate CSV: new search generates a report from a new search with data: default 1`
         │
         │ - Snapshot  - 741
         │ + Received  + 883
         │
```
<img width="1728" alt="Screenshot 2023-12-19 at 09 43 01" src="https://github.com/elastic/kibana/assets/4858051/dd6c9cda-c66d-45c8-b924-a989139faf58">

# Locally Updating the Snapshot works, but it still fails on the subsequent test run
```
         └- ✖ fail: discover Discover CSV Export Generate CSV: new search generates a report from a new search with data: default
         │      Error: expect(received).toMatchSnapshot()
         │
         │ Snapshot name: `discover Discover CSV Export Generate CSV: new search generates a report from a new search with data: default 1`
         │
         │ - Snapshot  - 238
         │ + Received  + 238
         │
         │ @@ -1,33 +1,6 @@
```


@kertal notice how the **Snapshot** vs **Received** numbers are vastly different in the original form, compared to the updated snapshot         

Beyond this, I've no idea what to do next.  

@kertal one other thing is that the name of the test is: `generates a report from a new search with data: default`, so I guess I'm not sure what the snapshot assertion is buying beyond the two assertions before it.  Your thoughts?

Reply:
just quick feedback, can take a closer look at it tomorrow.
you first screenshot is referring to a records with the _id **YWMtOWOBH63Xcmy453H9**
in the snapshot is can be found here:
https://github.com/elastic/kibana/blob/5070b9fd0372e4c55e4a9d8300c4a2f48f9c7214/x-pack/test/functional/apps/discover/__snapshots__/reporting.snap#L1942
according to the snapshot the first record should have the id **9AMtOW0BH63Xcmy432DJ**

Now I'd suggest to have a closer look why this expect failed, is the sort order (order_date) different? are there multiple timestamps with the same value (if you insert concurrently, for these records the order changes). If this is the case, there should be a 2nd sort param introduced, the `order_id` might be a good fit for this. In any case the snapshot needs to be updated, however, the sort oder should be stabilized (there might be another reason for this failure ... this is my first guess)

Reply:
I made some progress, adding another sort param here:

https://github.com/elastic/kibana/blob/53e803e42c5c9de73e60ae95cff13d17bab09f26/x-pack/test/functional/fixtures/kbn_archiver/reporting/ecommerce.json#L442-L446

adding `order_id` leads to stable sort ... but in one case I get an empty export now ... 👿 
 ```
[
        "order_date",
        "desc"
      ],
      [
        "order_id",
        "desc"
      ]

```

Reply:
@kertal regarding the empty export, where does that show up?

Reply:
in the newly generated snap file (download_csv.snap)


```
exports[`dashboard Reporting Download CSV Default Saved Search Data Download CSV export of a saved search panel 1`] = `
"\\"order_date\\",category,currency,\\"customer_id\\",\\"order_id\\",\\"day_of_week_i\\",\\"products.created_on\\",sku
"
`;

```

there should be documents ... the other tests here were unstable without additional sorting, but now they are stable ... the size of the document should now change because of this PR

Reply:
I think I found a solution ... keep finders crossed

Reply:
Thanks for the assist man, I cant wait to get this merged lol 
After your review, Robert will review and then hopefully we can put this to bed

Reply:
need more fingers crossed, tested this PR, this fixed some issues and broker others: 83c26aeb845bb6ea0d4b4e247a7ebf8cd64e9003

Reply:
Sounds like life! lol

Reply:
Life said: 💚  #173896